src/java.base/linux/native/libnio/ch/FileDispatcherImpl.c

```diff
@@ -103,6 +105,7 @@ Java_sun_nio_ch_FileDispatcherImpl_transferTo0(JNIEnv *env, jobject this,
                 case EINVAL:
                 case ENOSYS:
                 case EXDEV:
+                case EPERM:
```

Comment:
This is unrelated to [JDK-8316304](https://bugs.openjdk.org/browse/JDK-8316304). If you want to fix this as well, then please as a separate patch.